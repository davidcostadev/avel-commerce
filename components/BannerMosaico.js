import React from 'react'
import Link from 'next/link'
import styled from 'styled-components'
import styles from '../assets/scss/App.scss'
import theme from '../layout/theme'

const BannerMosaicoGrid = styled.div`
  display: flex;
  flex-wrap: wrap;
`
const BannerMosaicoCol = styled.div`
  flex: 1 1 100%;
  display: flex;
  justify-content: stretch;
  align-items: stretch;
  flex-direction: column;

  @media(min-width: 768px) {
    flex: 1 1 50%;
    max-width: 50%;
  }
`
const BannerMosaicoRow = styled.div`
  flex: 1 1 auto;
  display: flex;
  width: 100%;
  overflow: hidden;
`

const BannerText1 = styled.div`
  flex-basis: auto;
  color: ${theme.colorPrimary};
  font-size: 18px;
  margin-bottom: 20px;
  transition: transform 0.6s ease;
`
const BannerText2 = styled.div`
  flex-basis: auto;
  color: ${theme.white};
  font-size: 35px;
  margin-bottom: 20px;
  transition: transform 0.6s ease;
`
const BannerBtn = styled.a`
  flex-basis: auto;
  background-color: ${theme.colorPrimary};
  padding: 8px 12px;
  font-size: 18px;
  color: ${theme.white} !important;
  margin-bottom: 20px;
  border-width: 0;
  transition: background-color 0.4s ease;
  transition: transform 0.6s ease;

  &:hover {
    background-color: ${theme.colorPrimary}
  }
`

const BannerToCenter = `
  align-items: center;
  justify-content: center;
  ${BannerText1},
  ${BannerText2},
  ${BannerBtn} {
    text-align: center;
  }
`

const BannerToRight = `
  align-items: flex-end;
  ${BannerText1},
  ${BannerText2},
  ${BannerBtn} {
    text-align: right;
  }
`
const BannerToLeft = `
  align-items: flex-start;
  ${BannerText1},
  ${BannerText2},
  ${BannerBtn} {
    text-align: left;
  }
`

const BannerMosaicoItem = styled.div`
  flex: 1 1 50%;
  display: flex;
  flex-direction: column;
  background-size: cover;
  background-position: center;
  padding: 40px 60px;
  transition: transform 0.4s ease;

    &:hover {
    transform: scale(1.1);

    ${BannerText1},
    ${BannerText2},
    ${BannerBtn} {
      transform: scale(1.05)
    }
  }

  ${props => props.center ? BannerToCenter : ''}
  ${props => props.right ? BannerToRight : ''}
  ${props => props.left ? BannerToLeft : ''}
`


const BannerMosaico = () => (
  <BannerMosaicoGrid>
    <BannerMosaicoCol className={styles.bannerMosaicoCol}>
      <BannerMosaicoRow className={styles.bannerMosaicoRow}>
        <BannerMosaicoItem center className={`${styles.bannerMosaico} ${styles.bannerMosaicoCenter}`} style={{ backgroundImage: 'url(\'/static/img/phone.jpg\')' }}>
          <BannerText1 className={styles.bannerText1}>Celulares</BannerText1>
          <BannerText2 className={styles.bannerText2}>Tecnologia de Ponta</BannerText2>
          <Link href="/product">
            <BannerBtn className={styles.bannerBtn}>Ver Ofertas</BannerBtn>
          </Link>
        </BannerMosaicoItem>
      </BannerMosaicoRow>
    </BannerMosaicoCol>
    <BannerMosaicoCol className={styles.bannerMosaicoCol}>
      <BannerMosaicoRow className={styles.bannerMosaicoRow}>
        <BannerMosaicoItem left className={`${styles.bannerMosaico} ${styles.bannerMosaicoLeft}`} style={{ backgroundImage: 'url(\'/static/img/notebook.jpg\')' }}>
          <BannerText1 className={styles.bannerText1}>Notebook</BannerText1>
          <BannerText2 className={styles.bannerText2}>Ultima Geração</BannerText2>
          <Link href="/product">
            <BannerBtn className={styles.bannerBtn}>Ver Ofertas</BannerBtn>
          </Link>
        </BannerMosaicoItem>
      </BannerMosaicoRow>
      <BannerMosaicoRow className={styles.bannerMosaicoRow}>
        <BannerMosaicoItem right className={`${styles.bannerMosaico} ${styles.bannerMosaicoRight}`} style={{ backgroundImage: 'url(\'/static/img/perfume.jpg\')' }}>
          <BannerText1 className={styles.bannerText1}>Perfumes</BannerText1>
          <BannerText2 className={styles.bannerText2}>Melhores Marcas</BannerText2>
          <Link href="/product">
            <BannerBtn className={styles.bannerBtn}>Ver Ofertas</BannerBtn>
          </Link>
        </BannerMosaicoItem>
      </BannerMosaicoRow>
    </BannerMosaicoCol>
  </BannerMosaicoGrid>
)

export default BannerMosaico
