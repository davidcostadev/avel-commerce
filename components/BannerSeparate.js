import React from 'react'
import Link from 'next/link'
import styles from '../assets/scss/App.scss'

const BannerSeparate = () => (
  <div className={styles.bannerSeparateGrid}>
    <div className={styles.bannerSeparateCol}>
      <div className={styles.bannerSeparateRow}>
        <div className={`${styles.bannerSeparate} ${styles.bannerSeparateRight}`} style={{ backgroundImage: 'url(\'/static/img/camera.jpg\')' }}>
          <span className={styles.bannerText1}>Limpador de Tela</span>
          <span className={styles.bannerText2}>R$ 17,90</span>
          <Link href="/product">
            <a className={styles.bannerBtn}>Ver Detalhes</a>
          </Link>
        </div>
      </div>
    </div>
    <div className={styles.bannerSeparateCol}>
      <div className={styles.bannerSeparateRow}>
        <div className={`${styles.bannerSeparate} ${styles.bannerSeparateRight}`} style={{ backgroundImage: 'url(\'/static/img/mouse.jpg\')' }}>
          <span className={styles.bannerText1}>Mouse Mac</span>
          <span className={styles.bannerText2}>R$ 227,90</span>
          <Link href="/product">
            <a className={styles.bannerBtn}>Ver Detalhes</a>
          </Link>
        </div>
      </div>
    </div>
  </div>
)

export default BannerSeparate
