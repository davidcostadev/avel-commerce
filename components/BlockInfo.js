import React from 'react'
import PropTypes from 'prop-types'
import styles from '../assets/scss/App.scss'

const contentEscape = content => ({
  __html: content,
})

const BlockInfo = ({ content }) => (
  <section className={styles.blockInfo}>
    <header className={styles.blockInfoHeader}>
      <h2>Informações do Produto</h2>
    </header>
    <div className={styles.blockInfoContent} dangerouslySetInnerHTML={contentEscape(content)} />
  </section>
)

BlockInfo.propTypes = {
  content: PropTypes.string.isRequired,
}

export default BlockInfo
