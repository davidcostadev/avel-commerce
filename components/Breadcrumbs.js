import React from 'react'
import { Link } from '../routes'
import classNames from 'classnames'

import styles from '../assets/scss/App.scss'

const BreadcrumbItemLink = ({ item }) => (
  <li className={classNames([styles.breadcrumbItem, 'breadcrumb-item'])}>
    <Link route={item.route}><a>{item.title}</a></Link>
  </li>
)
const BreadcrumbItemCurrent = ({ item }) => (
  <li className={classNames([styles.breadcrumbItem, 'breadcrumb-item', 'active'])}>
    {item.title}
  </li>
)


const Breadcrumbs = ({ itens }) => (
  <ol className={classNames([styles.breadcrumb, 'breadcrumb'])}>
    {itens.map((item, index) => {
      console.log(itens.length - 1 === index)
      if (itens.length - 1 === index) {
        return <BreadcrumbItemCurrent item={item} key={index}/>
      }

      return <BreadcrumbItemLink item={item} key={index} />
    })}
  </ol>
)

export default Breadcrumbs
