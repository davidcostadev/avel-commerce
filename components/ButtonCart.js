import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Link } from '../routes'
import styles from '../assets/scss/App.scss'
import theme from '../layout/theme'

const Badge = styled.span`
  display: inline-block;
  width: 22px;
  height: 22px;
  padding: 5px 0px;
  background: ${theme.colorSecond};
  border-radius: 50%;
  font-size: 12px;
  vertical-align: super;
  color: white;
  transform: translate(4px, -8px);
`

const ButtonCartWrap = styled.a`
  display: flex;
  align-items: center;
  color: ${theme.colorPrimary} !important;
  background-color: white;
  border-radius: 3px;
  border-width: 0;
  padding: 0 10px;
  cursor: pointer;

  &:focus,
  &:hover {
    background-color: ${theme.buttonDefault} !important;
    color: green;
  }
  &:active {
    background-color: ${theme.buttonDefaultHover} !important;
  }
  i {
    font-size: 32px;
    vertical-align: middle;
  }
`

export const ButtonCart = ({ quant }) => (
  <Link route="/cart">
    <ButtonCartWrap>
      <i className="ion-ios-cart-outline" />
      <Badge className={`badge ${styles.badge}`}>{quant}</Badge>
    </ButtonCartWrap>
  </Link>
)

const ButtonCartStyled = styled(ButtonCart)`

`


ButtonCart.propTypes = {
  quant: PropTypes.number,
}

ButtonCart.defaultProps = {
  quant: 0,
}

const mapState = state => ({
  quant: state.cartItens.length,
})

export default connect(mapState)(ButtonCartStyled)
