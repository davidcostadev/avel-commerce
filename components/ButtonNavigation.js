import React from 'react'
import { connect } from 'react-redux'
import classNames from 'classnames'
import PropsTypes from 'prop-types'
import { toggleMenu } from '../flux/config/configActions'

import {
  MenuButton,
} from '../layout/Header'

const ButtonNavigation = ({ menuMobile, toogleMenu }) => {
  const classList = {
    'ion-navicon': !menuMobile,
    'ion-ios-close-empty': menuMobile,
  }

  return (
    <MenuButton onClick={toogleMenu}>
      <i className={classNames(classList)} />
    </MenuButton>
  )
}

ButtonNavigation.propTypes = {
  menuMobile: PropsTypes.bool.isRequired,
  toogleMenu: PropsTypes.func.isRequired,
}

const mapState = state => state

const mapDispatchToProps = dispatch => ({
  toogleMenu: () => dispatch(toggleMenu()),
})

export default connect(mapState, mapDispatchToProps)(ButtonNavigation)
