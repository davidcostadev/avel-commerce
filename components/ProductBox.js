import React from 'react'
import PropTypes from 'prop-types'
import { Link } from '../routes'
import styles from '../assets/scss/App.scss'
import { StringToReal } from '../utils/money'
import SquareBox from './SquareBox'
import { parcelado } from './ParceladoBox'


export const ProductBox = ({ product }) => (
  <div className={styles.product}>
    <SquareBox image={product.PS_PATH_IMAGEM_250} />
    {/* <picture className={styles.productImage}>
      <img src={product.PS_PATH_IMAGEM_250} alt={product.PS_PRODUTO} />
    </picture> */}
    <div className={styles.productTitle}>{product.PS_PRODUTO}</div>
    <div className={styles.productPrice}>
      <span className={styles.currency}>R$</span>
      <span className={styles.amount}>{StringToReal(product.PS_VALOR_DE_VENDA)}</span>
    </div>
    {parcelado(product.PS_VL_VENDA_CCCREDITO3X)}
    <div className={styles.productButtons}>
      <Link route={`/product/${product.PS_PATH_PAGE}`}>
        <a className={`btn ${styles.btn} ${styles.btnDetail}`}>Ver Detalhes</a>
      </Link>
      <Link route="/product">
        <a className={`btn ${styles.btn} ${styles.btnBuy}`}>Comprar</a>
      </Link>
    </div>
  </div>
)

ProductBox.propTypes = {
  product: PropTypes.object.isRequired,
}


export default ProductBox
