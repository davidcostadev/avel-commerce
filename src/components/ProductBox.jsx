import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from '../assets/scss/App.scss';

function toReal(number) {
  const float = parseFloat(number.replace(',', '.'), 10);
  return float.toLocaleString('pt-BR', { minimumFractionDigits: 2 });
}

const ProductBox = ({ product }) => (
  <div className={styles.product}>
    <picture className={styles.productImage}>
      <img src={product.PS_PATH_IMAGEM_250} alt={product.PS_PRODUTO}ss />
    </picture>
    <div className={styles.productTitle}>{product.PS_PRODUTO}</div>
    <div className={styles.productPrice}>
      <span className={styles.currency}>R$</span>
      <span className={styles.amount}>{toReal(product.PS_VALOR_DE_VENDA)}</span>
    </div>
    <div className={styles.productButtons}>
      <Link to={`/product/${product.PS_PATH_PAGE}`} className={`btn ${styles.btn} ${styles.btnDetail}`}>Ver Detalhes</Link>
      <Link to="/product" className={`btn ${styles.btn} ${styles.btnBuy}`}>Comprar</Link>
    </div>
  </div>
);

ProductBox.propTypes = {
  product: PropTypes.object.isRequired
};


export default ProductBox;
