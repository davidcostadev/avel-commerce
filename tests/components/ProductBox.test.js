/* eslint no-undef: "off" */
import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { ParcelBox, ProductBox } from '../../components/ProductBox'

const product = {
  PS_REGISTROS_COUNT: '5027',
  PS_ID_PRODUTO: '45093',
  PS_PRODUTO: 'Gabinete Slim DT-100BK C/Fonte PS-200 FX C3TECH S/Cabo',
  PS_VALOR_DE_VENDA: '284',
  PS_PRECO_SEMDESC: '284',
  PS_VL_VENDA_DESCDINH: '284',
  PS_VL_VENDA_CCDEBITO: '284',
  PS_VL_VENDA_CCREDITO: '284',
  PS_VL_VENDA_CCCREDITO1X: '284',
  PS_VL_VENDA_CCCREDITO3X: '284',
  PS_VL_VENDA_CCCREDITO6X: '314,4',
  PS_VL_VENDA_DESCONTO: '301',
  PS_TX_DESCONTO_PRODUTO: '0',
  PS_TX_DESCONTO_BOLETO: '6',
  PS_TX_DESCONTO_DINHEIRO: '0',
  PS_ESTOQUE_LOJA: '14',
  PS_ESTOQUE_FILIAL: '0',
  PS_ESTOQUE_DEPOSITO: '0',
  PS_ESTOQUE_TOTAL: '14',
  PS_DISPONIBILIDADE: 'Entrega Imediata',
  PS_FAMILIA: 'Informática',
  PS_GRUPO: 'Gabinete',
  PS_SUBGRUPO: 'Gamer',
  PS_ID_FAMILIA: '187',
  PS_ID_GRUPO: '217',
  PS_ID_SUBGRUPO: '2413',
  PS_FLAG_IMPORTADO: '2',
  PS_FLAG_PROMOCAO: '0',
  PS_FLAG_DISPONIBILIDADE: '1',
  PS_FLAG_LANCAMENTO: '1',
  PS_IMPORTADO: 'NÃO',
  PS_PROMOCAO: '',
  PS_LANCAMENTO: '',
  PS_PESO_GR: '3000',
  PS_COMPRIMENTO_MM: '0',
  PS_LARGURA_MM: '0',
  PS_ALTURA_MM: '0',
  PS_DIAMETRO_MM: '1',
  PS_ID_FORMATO_ENVIO: '0',
  PS_FABRICANTE: 'NONE',
  PS_MARCA: 'NONE',
  PS_UNIDADE: 'PC',
  PS_QT_UNITVENDA: '1',
  PS_TEMPODEGARANTIA_MES: '12',
  PS_DESCRICAO_VENDA: 'Característica:\r\n\r\nMarca: C3 Tech\r\nModelo: DT-100BK',
  PS_GARANTIA: '12 MESES',
  PS_ID_IMAGEM: '193420',
  PS_PATH_IMAGEM_150: 'http://www.winerp.com.br/images/mundial/products/gabinete-slim-dt-100bk-c-fonte-ps-200-fx-c3tech---45093-150-193420.jpg',
  PS_PATH_IMAGEM_250: 'http://www.winerp.com.br/images/mundial/products/gabinete-slim-dt-100bk-c-fonte-ps-200-fx-c3tech---45093-250-193420.jpg',
  PS_IMAGEM_TXT_ALT: '',
  PS_IMAGEM_TXT_TITLE: '',
  PS_PATH_PAGE: 'gabinete-slim-dt-100bk-c-fonte-ps-200-fx-c3tech',
  PS_LINK_TITLE: 'Gabinete Slim DT-100BK C/Fonte PS-200 FX C3TECH em Ribeirão Preto',
}

describe('Test File ProductBox', () => {
  it('<ProductBox />', () => {
    const create = () => {
      const store = {
        getState: jest.fn(() => ({})),
        dispatch: jest.fn(),
        subscribe: jest.fn(),
      }
      const next = jest.fn()

      const invoke = action => thunk(store)(next)(action)

      return { store, next, invoke }
    }
    const { store } = create()

    const tree = renderer.create((
      <Provider store={store}>
        <ProductBox
          product={product}
        />
      </Provider>
    )).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
