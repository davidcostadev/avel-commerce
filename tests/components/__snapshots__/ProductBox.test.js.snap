// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test File ProductBox <ProductBox /> 1`] = `
<div
  className="App__product___3XoPp"
>
  <picture
    className="SquareBox__BoxSquare-s7lvzn-0 gcTDzr"
  >
    <div
      className="SquareBox__BoxSquareInner-s7lvzn-1 UfKUk"
    />
  </picture>
  <div
    className="App__productTitle___3wgab"
  >
    Gabinete Slim DT-100BK C/Fonte PS-200 FX C3TECH S/Cabo
  </div>
  <div
    className="App__productPrice___ZoumY"
  >
    <span
      className="App__currency___2-n_1"
    >
      R$
    </span>
    <span
      className="App__amount___25x4z"
    >
      284,00
    </span>
  </div>
  <div
    className="App__productComplement___3_X5C"
  >
    ou 
    3
    x de
    <span
      className="App__currency___2-n_1"
    >
      R$
    </span>
    <span
      className="App__amount___25x4z"
    >
      94,67
    </span>
    <strong>
      Sem Juros
    </strong>
  </div>
  <div
    className="App__productButtons___2Y3pI"
  >
    <a
      className="btn App__btn___1w-MS App__btnDetail___13UmL"
      href="/product/gabinete-slim-dt-100bk-c-fonte-ps-200-fx-c3tech"
      onClick={[Function]}
    >
      Ver Detalhes
    </a>
    <a
      className="btn App__btn___1w-MS App__btnBuy___2ICLf"
      href="/product"
      onClick={[Function]}
    >
      Comprar
    </a>
  </div>
</div>
`;
